activesupport/lib/active_support/xml_mini/rexml.rb

```diff
@@ -20,11 +20,9 @@ def parse(data)
         data = StringIO.new(data || '')
       end
 
-      char = data.getc
-      if char.nil?
+      if data.eof?
```

Comment:
Does `data.eof?` mean we have no data?
Some data might already have been read and we might falsely consider it as eof? ?